clients/java/signalr/src/main/java/InvocationMessage.java

```diff
@@ -1,7 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-public class InvocationMessage {
+public class InvocationMessage extends HubMessage {
     int type = 1;
```

Comment:
This should now be of `HubMessageType`

Reply:
`HubMessageType` is just the enum tied to each of the specific message types (InvocationMessage, StreamItem etc). The actual base class is HubMessage

Reply:
? `type` should be `HubMessageType` instead of `int`

Reply:
Oh, my comment was referring to line 4, now realizing you were talking about line 5

Reply:
Yeah, was going to give a long explanation about why this is the way it is, but I think I'll be able to change it once I give the enums values. That might not work how I expect though since enums in java are slightly more complex

Reply:
Essentially I had the type as an int field for serialization. Then an abstract method on the base type `HubMessage` which is used at the `HubConnection` level